crates/orderbook-commons/src/lib.rs

```diff
@@ -21,7 +21,11 @@ pub use crate::price::Price;
 pub use crate::price::Prices;
 
 /// The prefix used in the description field of an order-matching fee invoice to be paid by a taker.
-pub const FEE_INVOICE_DESCRIPTION_PREFIX_TAKER: &str = "taker-fee-";
+pub const FEE_INVOICE_DESCRIPTION_PREFIX_TAKER: &str = "taker-fee";
```

Comment:
📣 I appreciate that it's probably nicer to remove this dash, but I think this is a breaking change. Probably worth avoiding that headache.

As a side note, I am pretty sure the prefix idea that I introduced is not very good and there were better alternatives 🙈

Reply:
> 📣 I appreciate that it's probably nicer to remove this dash, but I think this is a breaking change. Probably worth avoiding that headache.

Yea, most definitely :sweat_smile: 

Maybe for now I should add a legacy match.

The reason I added this was because there isn't really a way to match on multiple prefixes cleanly, whereas if you define the syntax for a structured payment invoice description as 

```
{tag} {additional_data}
```

then you can just `split_once` by space and match on the tag to get what kind of invoice it is.

However, it's worth noting that both this implementation _and_ the previous implementation could lead to some strange results. 

For instance, what would happen if an invoice were paid by a mischievous (or even malicious) user with the description `taker-fee-NotAnOrderId?!/.sidfj`? At best this would display wrong, worse it could crash the app, and at worst it could introduce a security vulnerability.

It's probably worth workshopping how this should work :sweat_smile: